apps/native-component-list/src/screens/Clipboard/ClipboardScreen.tsx

```diff
@@ -0,0 +1,212 @@
+
+const HAS_X_ASYNC_CONFIG: FunctionDescription = {
+  name: 'hasXAsync',
+  parameters: [],
+  actions: [
+    {
+      name: 'hasStringAsync',
+      action: Clipboard.hasStringAsync,
+    },
+    {
+      name: 'hasImageAsync',
+      action: withSupportedPlatforms(['android', 'ios'], Clipboard.hasImageAsync),
+    },
+    {
+      name: 'hasUrlAsync',
+      action: withSupportedPlatforms(['ios'], Clipboard.hasImageAsync),
+    },
+  ],
+};
+
+const SET_STRIING_ASYNC_CONFIG: FunctionDescription = {
+  name: 'setStringAsync',
+  parameters: [
+    {
+      name: 'inputString',
+      type: 'enum',
+      values: [
+        { name: '"Hello, world!"', value: 'Hello, world!' },
+        { name: '"<p>HTML paragraph</p>"', value: '<p>HTML paragraph</p>' },
+        { name: '"<ul><li>HTML list</li></ul>"', value: '<ul><li>HTML list</li></ul>' },
+      ],
+    },
+    {
+      name: 'options',
+      type: 'object',
+      properties: [
+        {
+          name: 'inputFormat',
+          type: 'enum',
+          platforms: ['ios', 'android'],
+          values: [
+            { name: 'StringFormat.PLAIN_TEXT', value: Clipboard.StringFormat.PLAIN_TEXT },
+            { name: 'StringFormat.HTML', value: Clipboard.StringFormat.HTML },
+          ],
+        },
+      ],
+    },
+  ],
+  actions: (value: string, options: { inputFormat: Clipboard.StringFormat }) =>
+    Clipboard.setStringAsync(value, options),
+};
+
+const GET_STRING_ASYNC_CONFIG: FunctionDescription = {
+  name: 'getStringAsync',
+  parameters: [
+    {
+      name: 'options',
+      type: 'object',
+      properties: [
+        {
+          name: 'preferredFormat',
+          type: 'enum',
+          platforms: ['ios', 'android'],
+          values: [
+            { name: 'StringFormat.PLAIN_TEXT', value: Clipboard.StringFormat.PLAIN_TEXT },
+            { name: 'StringFormat.HTML', value: Clipboard.StringFormat.HTML },
+          ],
+        },
+      ],
+    },
+  ],
+  actions: async (options: { preferredFormat: Clipboard.StringFormat }) => {
+    const result = await Clipboard.getStringAsync(options);
+    return result.length > 0 ? result : '[nothing]';
+  },
+};
+
+const SET_IMAGE_ASYNC_CONFIG: FunctionDescription = {
+  name: 'setImageAsync',
+  platforms: ['ios', 'android'],
+  parameters: [
+    {
+      name: 'imageBase64Data',
+      type: 'constant',
+      value: '[selected from image picker]',
+    },
+  ],
+  actions: async () => {
+    const { granted } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+    if (granted) {
+      const result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.Images,
+        base64: true,
+      });
+      if (!result.cancelled && result.base64) {
+        await Clipboard.setImageAsync(result.base64);
+        return 'Image copied to clipboard';
+      }
+    } else {
+      Alert.alert('Permission required!', 'You must allow accessing images in order to proceed.');
```

Comment:
I dislike this one 😞 
I would not use `Alert` or any other popup 🤔 
Let's go with simple rejection maybe for now?
We need to add mechanism for handling failing cases, like missing prerequisites or promise rejections. Ideally we could log more info in the console, like what's the actual method signature that failed, etc 😁 
When we add such mechanism these rejected function calls would be presented gracefully as well 💪

Reply:
For now, I'd just return this message to be displayed in the result box

Reply:
To be honest I don't really get that why you're against all alerts/popups. I agree that in many cases they would be annoying (especially for correct and expected results), but here this is more like a rare exception and in my opinion presenting an alert is much better UX than looking for console logs.